ibm_cloud_sdk_core/utils.py

```diff
@@ -54,29 +54,40 @@ def string_to_datetime(string):
     """
     return date_parser.parse(string)
 
-def get_authenticator_from_environment(service_name):
+def read_external_sources(service_name):
     """
-    Checks the credentials file and VCAP_SERVICES environment variable
+    Try to get config from external sources, with the following priority:
+    1. Credentials file(ibm-credentials.env)
+    2. Environment variables
+    3. VCAP Services(Cloud Foundry)
     :param service_name: The service name
-    :return: the authenticator
+    :return: dict
     """
-    authenticator = None
-    # 1. Credentials from credential file
+    config = {}
+
     config = read_from_credential_file(service_name)
-    if config:
-        authenticator = contruct_authenticator(config)
 
-    # 2. From env variables
-    if not authenticator:
+    if not config:
         config = read_from_env_variables(service_name)
-        if config:
-            authenticator = contruct_authenticator(config)
 
-    # 3. Credentials from VCAP
-    if not authenticator:
+    if not config:
         config = read_from_vcap_services(service_name)
-        if config:
-            authenticator = contruct_authenticator(config)
+
+    return config
+
+def get_authenticator_from_environment(service_name):
+    """
+    Try to get authenticator from external sources, with the following priority:
+    1. Credentials file(ibm-credentials.env)
+    2. Environment variables
+    3. VCAP Services(Cloud Foundry)
+    :param service_name: The service name
+    :return: the authenticator
+    """
+    authenticator = None
+    config = read_external_sources(service_name)
+    if config:
+        authenticator = contruct_authenticator(config)
```

Comment:
s/contruct/construct/

Reply:
Also, "construct_authenticator" could probably be private since it should only be called from inside the get_authenticator_from_environment public method.